### DISCUSSION TITLE
Lua syntax highlighting fail

### DISCUSSION THREAD

**Discussion by Kimapr** [community/community#124596] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

In https://github.com/Kimapr/kf2vm/blob/889bccbdecc637c3c78a4468e1e91ab9c599a16f/num.lua#L107-L154 and https://github.com/Kimapr/kf2vm/blob/889bccbdecc637c3c78a4468e1e91ab9c599a16f/vm.lua#L1-L488 you can see multiline comments extending far beyond where they are supposed to be.

(num.lua - the weird interleaving of comments here serves a purpose: it allows to switch between two codes by adding/removing a single character)
![image](https://github.com/community/community/assets/50977029/6740a7cb-2b87-4080-9853-1c4abf36482b)
(vm.lua)
![image](https://github.com/community/community/assets/50977029/7c63ec3a-b64f-4727-8756-d7ec9a6a5bf4)

It's somewhat excusable in num.lua, because weird nestings of multiline literals like that are known to confuse crappy syntax highlighters (like Discord's), but what happened to vm.lua is egregious and unacceptable, that's a perfectly normal way to do comments being broken. Even Discord, whose primary audience is *gamers*, not programmers, does it better!

(num.lua, still wrong)
![image](https://github.com/community/community/assets/50977029/7a9a513a-2fe8-4f57-99af-a2234206f3a7)
(num.lua in vim, correct)
![image](https://github.com/community/community/assets/50977029/dd8222ab-9ae8-4680-a41e-c15a525bde42)
(vm.lua) (correct)
![image](https://github.com/community/community/assets/50977029/df21344d-ed42-4c68-9ad6-8adacc497577)

What seems to happen is Github's highlighter first removes all regular single-line comments and only then processes multiline comments, which is wrong because singleline and multiline comments can interact with each other in complex ways. Even C has some of this, and the highlighter doesn't do this for C so I don't understand how this bug happened.
```c
/*
abcdef
//*/
abcdef
//*/
```
```lua
--[[
abcdef
--]]
abcdef
--]]
```

**Comment by Zirunis** [comment]
I noticed the same issue and was about to open a discussion. Will now simply pile on here. I think it is the same issue where the highlighter never finds the delimiter of the multiline comment.
As an example check the following page (not my code!): [https://github.com/davidm/lua-balanced/blob/master/luabalanced.lua](https://github.com/davidm/lua-balanced/blob/master/luabalanced.lua)
There is a comment from line 1 to 147, and then code from there to the end of the file. The syntax highlighting (as you can see) seems to actually make no difference between the two parts, highlighting the entire thing as if it were a comment.

**Comment by Kimapr** [comment]
Looks like Github fixed this. Nice
